client/asset/btc/spv.go

```diff
@@ -588,12 +531,15 @@ func (w *spvWallet) syncStatus() (*syncStatus, error) {
 	var synced bool
 	var blk *block
 	// Wallet address manager sync height.
-	if chainBlk.Timestamp.After(w.wallet.Manager.Birthday()) {
-		// After birthday, wallet address manager should be syncing.
+	if chainBlk.Timestamp.After(w.wallet.birthday()) {
+		// After birthday, wallet address manager should be syncing. Although
+		// block time stamps are not necessarily monotonically increasing, this
+		// is a reasonable condition at which the wallet's sync height should be
+		// consulted instead of the chain service's height.
 		walletBlock := w.wallet.syncedTo()
 		if walletBlock.Height == 0 {
-			// About to start, so just return last chain service height prior to
-			// wallet birthday.
+			// The wallet is about to start its sync, so just return the last
+			// chain service height prior to wallet birthday until it begins.
```

Comment:
Working as expected to keep the progress from jumping backwards when wallet starts to sync.  If you insert logs here and the other branches, you see a few wallet state notes in this "about to start" state:

```
2022-01-21 18:14:26.000 [TRC] CORE: notify: |DATA| (walletstate)
2022-01-21 18:14:36.001 [DBG] CORE[btc][SPV]: before birthday 2021-05-31 21:12:00 -0500 CDT
2022-01-21 18:14:36.001 [TRC] CORE: notify: |DATA| (walletstate)
2022-01-21 18:14:46.000 [DBG] CORE[btc][SPV]: about to start wallet
2022-01-21 18:14:46.000 [TRC] CORE: notify: |DATA| (walletstate)
2022-01-21 18:14:56.001 [DBG] CORE[btc][SPV]: about to start wallet
2022-01-21 18:14:56.001 [TRC] CORE: notify: |DATA| (walletstate)
2022-01-21 18:15:01.999 [DBG] CORE[btc][SPV]: chain = 2136807, wallet = 1979545, target = 2136807 (synced = false)
```

These logs are removed from this PR however.